gpusim.h

```diff
@@ -24,8 +24,9 @@ class GPUSimServer : public QObject
      * Start a GPUSimServer running on a local socket to communicate with
      * clients.  The server will contain the data in the provided .fsim file.
      * @param database_fname: .fsim file storing relevant data in binary format
+     * @param gpu_bitcount: If 0, then auto-calculate optimal value
      */
-    GPUSimServer(const QStringList& database_fnames);
+    GPUSimServer(const QStringList& database_fnames, int gpu_bitcount=0);
```

Comment:
I started doing this - but it's actually more painful to write it this way when constructing it, as I have to switch from just creating an object, to having a shared_ptr that I initialize with the different constructors based on an if statement.  Later instead of being:

```
int gpubitcount = 0;
// figure out if it's set
GPUSimServer(list, gpubitcount);
```

It'll have to be
```
int gpubitcount = 0; // or add another true/false variable as well
// figure out if it's set
unique_ptr<GPUSimServer> server;
if (gpubitcount) server = make_unique<GPUSimServer>(list, gpubitcount);
else  server = make_unique<GPUSimServer>(list);
```

I'm not sure this is a net readability win.  Am I missing something?